src/controllers/simplified-account/services/my-services.controller.test.js

```diff
@@ -0,0 +1,410 @@
+const { formattedPathFor } = require('@root/paths')
+const { formatSimplifiedAccountPathsFor } = require('@utils/simplified-account/format')
+const formatServicePathsFor = require('@utils/format-service-paths-for')
+
+const SERVICE_NAME = 'Rare coin authentication service'
+const SERVICE_EXTERNAL_ID = 'service-123-def'
+
+const mockResponse = sinon.spy()
+const mockGatewayAccountService = {
+  getGatewayAccountsByIds: sinon.stub().resolves({
+    32: new GatewayAccount(validGatewayAccount({
+      gateway_account_id: 32,
+      service_name: SERVICE_NAME,
+      type: 'test',
+      payment_provider: WORLDPAY
+    })),
+    33: new GatewayAccount(validGatewayAccount({
+      gateway_account_id: 33,
+      service_name: SERVICE_NAME,
+      type: 'test',
+      payment_provider: 'unsupported payment provider'
+    })),
+    34: new GatewayAccount(validGatewayAccount({
+      gateway_account_id: 34,
+      service_name: SERVICE_NAME,
+      type: 'test',
+      payment_provider: STRIPE
+    })),
+    35: new GatewayAccount(validGatewayAccount({
+      gateway_account_id: 35,
+      service_name: SERVICE_NAME,
+      type: 'test',
+      payment_provider: SANDBOX
+    })),
+    36: new GatewayAccount(validGatewayAccount({
+      gateway_account_id: 36,
+      service_name: SERVICE_NAME,
+      type: 'test',
+      payment_provider: SANDBOX
+    })),
+    37: new GatewayAccount(validGatewayAccount({
+      gateway_account_id: 37,
+      service_name: SERVICE_NAME,
+      type: 'live',
+      payment_provider: WORLDPAY
+    }))
+  })
+}
+const userServiceRoles = [
+  new ServiceRole({
+    role: {
+      name: 'admin',
+      permissions: []
+    },
+    service: validServiceResponse({
+      external_id: SERVICE_EXTERNAL_ID,
+      gateway_account_ids: [],
+      name: SERVICE_NAME,
+      service_name: {
+        en: SERVICE_NAME
+      },
+      current_go_live_stage: LIVE
+    })
+  })
+]
+
+const {
+  req,
+  res,
+  nextRequest,
+  call
+} = new ControllerTestBuilder('@controllers/simplified-account/services/my-services.controller')
+  .withStubs({
+    '@utils/response': { response: mockResponse },
+    '@services/gateway-accounts.service': mockGatewayAccountService
+  })
+  .build()
+
+describe('Controller: services/my-services.controller', () => {
+  describe('get', () => {
+    describe('for a non-degatewayed user', () => {
+      describe('for a service with only a worldpay test account', () => {
+        before(() => {
+          userServiceRoles[0].service.gatewayAccountIds = ['32']
+          nextRequest({
+            user: {
+              isDegatewayed: () => false,
+              serviceRoles: userServiceRoles
+            }
+          })
+          call('get')
+        })
+
+        it('should set isWorldpayTestService on the service', () => {
+          sinon.assert.calledWithMatch(mockResponse,
+            sinon.match.any,
+            sinon.match.any,
+            sinon.match.any,
```

Comment:
Might as well match on req, res, and 'simplified-account/services/my-services/index' here?

Reply:
I already test the presence and correctness of these keys in other tests so I've omitted them here in favour of making this test easier to reason about